### ISSUE TITLE
Include status field

### ISSUE THREAD

**Issue by olantrust** [nextcloud/tasks#8]
Steps to reproduce:
1. Create new task
2. Open detailed pane

Expected behaviour:
User expects to be able to set the status of a task for 
a) easy management of lots of tasks to see where they are all at
b) ability to see instantly what the status of an individual task is

Actual behaviour:
No status field is available, only a % percentage progress. Which is useful, but doesn't meet the needs of working out a task's status.

Proposed solution:
Have a Status field.

Implementation:
- Choices for status field: 1) Proposed [awaiting confirmation] 2) Not started 3) In progress 4) Awaiting input [i.e. from an event or another use] 5) Deferred 6) Completed 
- Automation: a User could benefit from the status field, without ever changing it manually, it would still help in their organisation of tasks.
Status 2) would be the default status when a task was created
Status 3) would be set automatically when a % progress was set
Status 6) would be set automatically when a task was marked completed
- If preferred, there could be a configurable option to hide this field (or alternatively hide it by default and have an option to show this).

Server configuration:
Nextcloud version: 10
Tasks version: 0.9.3

**Comment by eppfel** [comment]
As Tasks uses the CalDav server backend. There is a specification draft we could use https://www.ietf.org/rfc/rfc2445

- "NEEDS-ACTION"       ;Indicates to-do needs action.
- "COMPLETED"           ;Indicates to-do completed.
- "IN-PROCESS"          ;Indicates to-do in process of
- "CANCELLED"           ;Indicates to-do was cancelled.

Of course we could always introduce custom property. Maybe other todo applications introduce something similar we can copy.

```
4.8.1.11 Status

   Property Name: STATUS

   Purpose: This property defines the overall status or confirmation for
   the calendar component.

   Value Type: TEXT

   Property Parameters: Non-standard property parameters can be
   specified on this property.

   Conformance: This property can be specified in "VEVENT", "VTODO" or
   "VJOURNAL" calendar components.

   Description: In a group scheduled calendar component, the property is
   used by the "Organizer" to provide a confirmation of the event to the
   "Attendees". For example in a "VEVENT" calendar component, the
   "Organizer" can indicate that a meeting is tentative, confirmed or
   cancelled. In a "VTODO" calendar component, the "Organizer" can
   indicate that an action item needs action, is completed, is in
   process or being worked on, or has been cancelled. In a "VJOURNAL"
   calendar component, the "Organizer" can indicate that a journal entry
   is draft, final or has been cancelled or removed.

   Format Definition: The property is defined by the following notation:

     status     = "STATUS" statparam] ":" statvalue CRLF

     statparam  = *(";" xparam)

     statvalue  = "TENTATIVE"           ;Indicates event is
                                        ;tentative.
                / "CONFIRMED"           ;Indicates event is
                                        ;definite.
                / "CANCELLED"           ;Indicates event was
                                        ;cancelled.
        ;Status values for a "VEVENT"

     statvalue  =/ "NEEDS-ACTION"       ;Indicates to-do needs action.
                / "COMPLETED"           ;Indicates to-do completed.
                / "IN-PROCESS"          ;Indicates to-do in process of
                / "CANCELLED"           ;Indicates to-do was cancelled.
        ;Status values for "VTODO".

     statvalue  =/ "DRAFT"              ;Indicates journal is draft.
                / "FINAL"               ;Indicates journal is final.
                / "CANCELLED"           ;Indicates journal is removed.
        ;Status values for "VJOURNAL".

   Example: The following is an example of this property for a "VEVENT"
   calendar component:

     STATUS:TENTATIVE

   The following is an example of this property for a "VTODO" calendar
   component:

     STATUS:NEEDS-ACTION

   The following is an example of this property for a "VJOURNAL"
   calendar component:

     STATUS:DRAFT

**Comment by raimund-schluessler** [comment]
This was implemented with #655.
